api/insight/apimiddleware.go

```diff
@@ -69,13 +72,24 @@ func (c *insightApiContext) getBlockHashCtx(r *http.Request) (string, error) {
 
 // GetRawHexTx retrieves the ctxRawHexTx data from the request context. If not
 // set, the return value is an empty string.
-func (c *insightApiContext) GetRawHexTx(r *http.Request) (string, bool) {
+func (c *insightApiContext) GetRawHexTx(r *http.Request) (string, error) {
 	rawHexTx, ok := r.Context().Value(ctxRawHexTx).(string)
 	if !ok {
-		apiLog.Trace("Rawtx hex transaction not set")
-		return "", false
+		apiLog.Trace("hex transaction id not set")
+		return "", fmt.Errorf("hex transaction id not set")
+	}
+	serializedTx, err := hex.DecodeString(rawHexTx)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode tx: %v", err)
 	}
-	return rawHexTx, true
+	msgtx := wire.NewMsgTx()
+	err = msgtx.Deserialize(bytes.NewReader(serializedTx))
+	if err != nil {
+		return "", fmt.Errorf("failed to deserialize tx: %v", err)
+	}
```

Comment:
The above 7 or so lines could be trimmed to:

```go
err := msgtx.Deserialize(hex.NewDecoder(strings.NewReader(rawHexTx)))
if err != nil {
	return "", fmt.Errorf("failed to deserialize tx: %v", err)
}
```

Unless the "failed to decode" adds important context not already gleaned from the DecodeString error.